core/procedures.js

```diff
@@ -358,9 +417,58 @@ Blockly.Procedures.editProcedureCallback_ = function(block) {
       return;
     }
     block = innerBlock;
+  } else if (!block.type == Blockly.PROCEDURES_PROTOTYPE_BLOCK_TYPE){
+    // This is a call block, find the prototype corresponding to the procCode
+    block = Blockly.Procedures.getPrototypeBlock(
+      block.getProcCode(), block.workspace);
   }
-  alert('TODO(#603): implement function editing (procCode was "' +
-      block.procCode_ + '")');
+
+  // Block now refers to the procedure prototype block, it is safe to proceed.
+  Blockly.Procedures.externalProcedureDefCallback_(
+    block.mutationToDom(),
+    Blockly.Procedures.editProcedureCallbackFactory_(block)
+  );
+};
+
+/**
+ * Callback factory for editing an existing custom procedure.
+ * @param {!Blockly.Block} block The procedure prototype block being edited.
+ * @return {function(?Element)} Callback for editing the custom procedure.
+ * @private
+ */
+Blockly.Procedures.editProcedureCallbackFactory_ = function(block) {
+  return function(mutation) {
+    if (mutation) {
+      // Update all the callers
+      var defineBlock = Blockly.Procedures.getDefineBlock(block.getProcCode(),
+        block.workspace);
+      if (defineBlock) {
+        var callers = Blockly.Procedures.getCallers(block.getProcCode(),
```

Comment:
This use of getCallers is kind of strained, since you had to go find the defineBlock even though you want to search in the entire workspace.  I added https://github.com/LLK/scratch-blocks/issues/1264 so we can circle back and clean it up later.